src/gbinder_remote_request.c

```diff
@@ -183,8 +183,6 @@ gbinder_remote_request_set_data(
             /* No RPC header */
             self->header_size = 0;
         }
-    } else {
```

Comment:
I think it's ok to simply remove the check for null  (and `gbinder_remote_request_set_data(NULL, 0, NULL)` from `unit_remote_request.c` test). The caller always unconditionally allocates `GBinderRemoteRequest` prior to calling this function, the `req` pointer is never null.

Reply:
#127 is what I had in mind. That covers `gbinder_remote_reply_set_data()` case too. It was easier for me to write a PR than to explain what I mean)